atomicapp/providers/kubeconfig.py

```diff
@@ -0,0 +1,118 @@
+import anymarkup
+
+from atomicapp.plugin import ProviderFailedException
+from atomicapp.constants import (ACCESS_TOKEN_KEY,
+                                 LOGGER_DEFAULT,
+                                 NAMESPACE_KEY,
+                                 PROVIDER_API_KEY,
+                                 PROVIDER_TLS_VERIFY_KEY,
+                                 PROVIDER_CA_KEY)
+import logging
+logger = logging.getLogger(LOGGER_DEFAULT)
+
+
+class KubeConfig(object):
+
+    @staticmethod
+    def parse_kubeconf(filename):
+        """"
+        Parse kubectl config file
+
+        Args:
+            filename (string): path to configuration file (e.g. ./kube/config)
+
+        Returns:
+            dict of parsed values from config
+
+        Example of expected file format:
+            apiVersion: v1
+            clusters:
+            - cluster:
+                server: https://10.1.2.2:8443
+                certificate-authority: path-to-ca.cert
+                insecure-skip-tls-verify: false
+              name: 10-1-2-2:8443
+            contexts:
+            - context:
+                cluster: 10-1-2-2:8443
+                namespace: test
+                user: test-admin/10-1-2-2:8443
+              name: test/10-1-2-2:8443/test-admin
+            current-context: test/10-1-2-2:8443/test-admin
+            kind: Config
+            preferences: {}
+            users:
+            - name: test-admin/10-1-2-2:8443
+            user:
+                token: abcdefghijklmnopqrstuvwxyz0123456789ABCDEF
+        """
+        logger.debug("Parsing %s", filename)
+
+        with open(filename, 'r') as fp:
+            kubecfg = anymarkup.parse(fp.read())
+
+        try:
+            return KubeConfig.parse_kubeconf_data(kubecfg)
+        except ProviderFailedException:
+            raise ProviderFailedException('Invalid %s' % filename)
+
+    @staticmethod
+    def parse_kubeconf_data(kubecfg):
```

Comment:
Why did we remove the _ ? It's still a method that should only be called from this file right?